controllers/storagecluster/initialization_reconciler_test.go

```diff
@@ -176,22 +281,23 @@ func initStorageClusterResourceCreateUpdateTestWithPlatform(
 
 	err := os.Setenv("OPERATOR_NAMESPACE", cr.Namespace)
 	assert.NoError(t, err)
-	result, err := reconciler.Reconcile(context.TODO(), request)
+	result, err := reconciler.Reconcile(context.TODO(), requestOCSInit)
 	assert.NoError(t, err)
 	assert.Equal(t, reconcile.Result{}, result)
 	err = os.Setenv("WATCH_NAMESPACE", cr.Namespace)
 	assert.NoError(t, err)
 
-	return t, reconciler, cr, request
+	return t, reconciler, cr, requestOCSInit
 }
 
 func createFakeInitializationStorageClusterReconciler(t *testing.T, obj ...runtime.Object) StorageClusterReconciler {
 	return createFakeInitializationStorageClusterReconcilerWithPlatform(
-		t, &Platform{platform: configv1.NonePlatformType}, obj...)
+		t, &Platform{platform: configv1.NonePlatformType}, nil, obj...)
 }
 
 func createFakeInitializationStorageClusterReconcilerWithPlatform(t *testing.T,
 	platform *Platform,
+	storageProfiles *api.StorageProfileList,
```

Comment:
Well, that would entail sending the `StorageProfile`s to `WithRuntimeObjects` contrary to my implementation that does _not_ treat those as runtime objects of `StorageCluster`. Furthermore, the cost of unpacking would be worse since I'd need to exclude the `StorageProfile`s from the set of runtime objects. So from my POV this exclusion would take place in either case and appending that function argument is the least costly option. Thoughts?

Reply:
> Well, that would entail sending the `StorageProfile`s to `WithRuntimeObjects` contrary to my implementation that does _not_ treat those as runtime objects of `StorageCluster`.

I'm proposing the exact opposite of this, which should be in keeping with the intent of your implementation. In my proposal we would still be *not* sending any StorageProfiles to `.WithRuntimeObjects()` and keeping them in the list to `.WithStatusSubresource()`. I do agree that the proposal is more computationally intensive, but the overhead would be minimal to negligible. This is a unit testing framework, and the addition of a few milliseconds sounds like a worthwhile sacrifice for the sake of cleaner implementations.

Naturally, this is only being suggested because we rely on a `...` parameter, the same reason you're adding a new parameter to begin with.

...all that said, it would probably be even easier to just rename the new parameter to `statusObjs` or something and do the StorageProfiles loop outside this function (since, again, it's only ever used once). Thus:

```suggestion
	statusObj []runtime.Object,
```

Followed by:

```
	statusObj = append(statusObj, sc)
```
Appending to a `nil` slice [works out fine](https://go.dev/play/p/Odak8oxqkcF). From a code clarity standpoint this seems more natural and descriptive than my earlier proposal, and less of a uni-tasker. I still don't like seeing a new parameter be largely useless except for one instance, but I can deal. :laughing:

Reply:
I like your 1st suggestion best 😅